app/selectors/accountsController.ts

```diff
@@ -249,3 +254,68 @@ export const selectSolanaAccount = createSelector(
 );
 
 ///: END:ONLY_INCLUDE_IF
+
+/**
+ * A memoized selector that returns all internal accounts that are valid for a given scope.
+ *
+ * For EVM scopes (eip155:*), this returns all accounts that have any EVM scope
+ * (i.e., any scope that starts with 'eip155:'). For non-EVM scopes, this returns
+ * all accounts that include the exact scope.
+ */
+export const selectInternalAccountsByScope = createDeepEqualSelector(
+  [
+    selectInternalAccountsById,
+    (_state: RootState, scope: CaipChainId) => scope,
+  ],
+  (
+    accountsMap: Record<AccountId, InternalAccount>,
+    scope: CaipChainId,
+  ): InternalAccount[] => {
+    const accounts = Object.values(accountsMap);
+    if (!Array.isArray(accounts) || accounts.length === 0) {
+      return [];
+    }
+
+    // Parse the requested scope
+    let namespace: string;
+    let reference: string;
+    try {
+      const parsed = parseCaipChainId(scope);
+      namespace = parsed.namespace;
+      reference = parsed.reference;
+    } catch {
+      return [];
+    }
+
+    if (namespace === KnownCaipNamespace.Eip155) {
+      const evmWildcard = `${KnownCaipNamespace.Eip155}:0` as CaipChainId;
+
+      // If requesting eip155:0 (wildcard), include any account that has any EVM scope
+      if (reference === '0') {
+        return accounts.filter(
+          (account) =>
+            Array.isArray(account.scopes) &&
+            account.scopes.some((s) =>
+              s.startsWith(`${KnownCaipNamespace.Eip155}:`),
+            ),
+        );
+      }
+
+      // For a specific EVM chain, include accounts that either:
+      // - have the exact scope (e.g., eip155:1), or
+      // - have the wildcard scope (eip155:0)
+      return accounts.filter(
+        (account) =>
+          Array.isArray(account.scopes) &&
+          (account.scopes.includes(scope) ||
+            account.scopes.includes(evmWildcard)),
```

Comment:
suggestion: you can remove `evmWildcard` and use the value `EthScope.Eoa` here from @metamask/keyring-api


```suggestion
            account.scopes.includes(EthScope)),
```

Reply:
Good suggestion.